### PR TITLE
Fix parameter types in ave_pool kernels, remove static_cast, add GLOBAL_MEM

### PR DESCRIPTION
**Pull request by xiaoqie**
Related to #6036

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins test this

File: theano/gpuarray/rng_mrg.py [hunk 1]
```diff
@@ -81,8 +81,8 @@ def gpu_kernels(self, node, name):
                 const ga_uint Nsamples,
                 const ga_uint Nstreams_used)
         {
-            sample_data = (GLOBAL_MEM %(otype)s *)(((char *)sample_data) + sample_offset);
-            state_data = (GLOBAL_MEM ga_int *)(((char *)state_data) + state_offset);
```
Comment (abergeron) [inline theano/gpuarray/rng_mrg.py]: This will need a version bump for the cache.

File: theano/gpuarray/multinomial.py [hunk 2]
```diff
@@ -297,9 +297,9 @@ def gpu_kernels(self, node, name):
     const ga_ssize outs_col_stride
 )
 {
-    global_pvals_copy = (GLOBAL_MEM float *)(((char *)global_pvals_copy) + global_pvals_offset);
-    global_unis = (GLOBAL_MEM float *)(((char *)global_unis) + global_unis_offset);
-    global_outs = (GLOBAL_MEM ga_long *)(((char *)global_outs) + global_outs_offset);
+    global_pvals_copy = (GLOBAL_MEM float *)(((GLOBAL_MEM char *)global_pvals_copy) + global_pvals_offset);
+    global_unis = (GLOBAL_MEM float *)(((GLOBAL_MEM char *)global_unis) + global_unis_offset);
+    global_outs = (GLOBAL_MEM ga_long *)(((GLOBAL_MEM char *)global_outs) + global_outs_offset);
```
Comment (abergeron) [inline theano/gpuarray/multinomial.py]: version bump

File: theano/gpuarray/multinomial.py [hunk 3]
```diff
@@ -84,9 +84,9 @@ def gpu_kernels(self, node, name):
     const ga_ssize outs_col_stride
 )
 {
-    global_pvals = (GLOBAL_MEM %(in_ctype)s *)(((char *)global_pvals) + global_pvals_offset);
-    global_unis = (GLOBAL_MEM %(in_ctype)s *)(((char *)global_unis) + global_unis_offset);
-    global_outs = (GLOBAL_MEM %(out_ctype)s *)(((char *)global_outs) + global_outs_offset);
+    global_pvals = (GLOBAL_MEM %(in_ctype)s *)(((GLOBAL_MEM char *)global_pvals) + global_pvals_offset);
+    global_unis = (GLOBAL_MEM %(in_ctype)s *)(((GLOBAL_MEM char *)global_unis) + global_unis_offset);
+    global_outs = (GLOBAL_MEM %(out_ctype)s *)(((GLOBAL_MEM char *)global_outs) + global_outs_offset);
```
Comment (abergeron) [inline theano/gpuarray/multinomial.py]: version bump

File: theano/gpuarray/basic_ops.py [hunk 4]
```diff
@@ -1632,7 +1632,7 @@ def gpu_kernels(self, node, name):
         code = """
 KERNEL void eye(GLOBAL_MEM %(ctype)s *a, ga_size a_off,
                 ga_size n, ga_size m, ga_ssize k) {
-    a = (GLOBAL_MEM %(ctype)s *)(((char *)a) + a_off);
+    a = (GLOBAL_MEM %(ctype)s *)(((GLOBAL_MEM char *)a) + a_off);
```
Comment (abergeron) [inline theano/gpuarray/basic_ops.py]: version bump

**Comment by xiaoqie** [comment]
@abergeron I have added the version bump, can you check it?

**Comment by nouiz** [comment]
jenkins test this

**Comment by xiaoqie** [comment]
Jenkins test errors don't seem related to this PR.

**Comment by abergeron** [comment]
I'll close and reopen to trigger a new merge.

**Comment by abergeron** [comment]
Jenkins is happy now, so we just need to wait for travis and we can merge.

**Comment by lamblin** [comment]
#6062 may help with the timeout.

**Comment by abergeron** [comment]
Since the problem with travis is a timeout that has been spotted on multiple other PRs, I'm ok to merge this.

**Pull request merged**